src/pages/workspaces/workspace/analysis/AnalysisLauncher.js

```diff
@@ -465,10 +468,17 @@ const AnalysisPreviewFrame = ({ analysisName, toolLabel, workspace: { workspace:
     withRequesterPaysHandler(onRequesterPaysError),
     withErrorReporting('Error previewing analysis')
   )(async () => {
-    const previewHtml = !!googleProject ?
+    // TOODO: Tracked in IA-4015. This implementation is not ideal. Introduce Error typing to better resolve the response.
+    const response = !!googleProject ?
       await Ajax(signal).Buckets.analysis(googleProject, bucketName, analysisName, toolLabel).preview() :
       await Ajax(signal).AzureStorage.blob(workspaceId, analysisName).preview()
-    setPreview(previewHtml)
+
+    if (response.status === 200) {
+      Metrics().captureEvent(Events.analysisPreviewSuccess, { fileName: analysisName, fileType: getExtension(analysisName), cloudPlatform: !!googleProject ? cloudProviderTypes.GCP : cloudProviderTypes.AZURE })
+    } else {
+      Metrics().captureEvent(Events.analysisPreviewFail, { fileName: analysisName, fileType: getExtension(analysisName), cloudPlatform: !!googleProject ? cloudProviderTypes.GCP : cloudProviderTypes.AZURE, errorText: response.statusText })
+    }
+    setPreview(await response.text())
```

Comment:
huh... I don't like having a nested `await`, maybe move this to the line above?

Otherwise LGTM

Reply:
Sure thing!